lib/rspec/mocks/example_methods.rb

```diff
@@ -201,6 +201,14 @@ def allow_message_expectations_on_nil
         RSpec::Mocks.space.proxy_for(nil).warn_about_expectations = false
       end
 
+      # Raise an error when setting expectations on nil.
+      #
+      # By default warning messages are issued when expectations are set on
+      # nil.  This is to force test failures when an expectation is set on nil.
+      def disallow_message_expectations_on_nil
```

Comment:
This mirrors the existing API (`allow_message_expectations_on_nil`) being something you set for a particular example...but that's always felt odd to me given it's the only config we handle that way.  I think many people are going to want  this set for all examples and to not have to set it in each example to achieve that.

WDYT about moving this to a global config option?

``` ruby
RSpec.configure do |config|
  config.mock_with :rspec do |mocks|
    mocks.allow_message_expectations_on_nil = false
  end
end
```